### PR TITLE
feat: marked questions shows in quiz

### LINKED ISSUE #146: show marked question in quiz sheet
**Describe the feature you'd like**
Show marked items in the quiz bottom similar to web

### PR DESCRIPTION
**Pull request by raystatic**
Fixes #146 

Changes: changed the implementation of choice options using recyclerview

Screenshots for the change:
WhatsApp Image 2019-05-15 at 15 09 44

### CODE REVIEW

File: app/src/main/java/com/codingblocks/cbonlineapp/adapters/ChoicesAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    private var question = ArrayList<Question>()
+
+    fun setdata(question: ArrayList<Question>, numbers:ArrayList<Int>){
+        this.numbers = numbers
+        this.question = question
+        notifyDataSetChanged()
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
+        context = parent.context
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/adapters/ChoicesAdapter.kt]: not needed

File: app/src/main/java/com/codingblocks/cbonlineapp/adapters/ChoicesAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    override fun getItemCount(): Int {
+        return question.size
+    }
+
+    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
+
+        holder.itemView.numberbtn.text = (position+1).toString()
+
+        Log.d("marks_rv","${numbers.size}")
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/adapters/ChoicesAdapter.kt]: remove logs

File: app/src/main/java/com/codingblocks/cbonlineapp/adapters/ChoicesAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+        for (i in numbers){
+            if (position == i){
+                holder.itemView.numberbtn.background = context.getDrawable(R.drawable.button_rounded_green_background)
+            }
+        }
+
+    }
+
+    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/adapters/ChoicesAdapter.kt]: whats point of creating ViewHolder if everything is done in the adapter itself

File: app/src/main/java/com/codingblocks/cbonlineapp/fragments/QuizFragment.kt [hunk 4]
```diff
@@ -81,44 +90,61 @@ class QuizFragment : Fragment(), AnkoLogger, ViewPager.OnPageChangeListener, Vie
...
+        choiceNumberAdapter.notifyDataSetChanged()
+    }
+
+    override fun onChoiceClicked(pos: Int) {
+        quizViewPager.currentItem = pos
+    }
+
+    private fun setUpQuestionBottomSheet(questions: ArrayList<Question>) {
+//        var count = 0
+//
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/fragments/QuizFragment.kt]: remove this

File: app/src/main/res/drawable/button_rounded_green_background.xml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<ripple xmlns:android="http://schemas.android.com/apk/res/android"
+    android:color="@color/white_transparency"> <!-- The ripple will be red -->
```
Comment (aggarwalpulkit596) [inline app/src/main/res/drawable/button_rounded_green_background.xml]: ripple will be red but white color?

File: app/src/main/java/com/codingblocks/cbonlineapp/fragments/QuizFragment.kt [hunk 6]
```diff
@@ -28,18 +34,21 @@ private const val ARG_ATTEMPT_ID = "attempt_id"
...
     private lateinit var quizId: String
     private lateinit var attemptId: String
     private lateinit var quizAttemptId: String
     private lateinit var firebaseAnalytics: FirebaseAnalytics
 
-
     lateinit var mAdapter: ViewPagerAdapter
     var questionList = HashMap<Int, String>()
     var sheetBehavior: BottomSheetBehavior<*>? = null
+    var markedpositions = ArrayList<Int>()
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/fragments/QuizFragment.kt]: remove these also

File: .idea/modules.xml [hunk 7]
```diff
@@ -2,7 +2,7 @@
 <project version="4">
   <component name="ProjectModuleManager">
     <modules>
-      <module fileurl="file://$PROJECT_DIR$/.idea/CBOnlineApp.iml" filepath="$PROJECT_DIR$/.idea/CBOnlineApp.iml" />
+      <module fileurl="file://$PROJECT_DIR$/CBOnlineApp.iml" filepath="$PROJECT_DIR$/CBOnlineApp.iml" />
```
Comment (aggarwalpulkit596) [inline .idea/modules.xml]: rebase your project and don't commit these files

File: app/src/main/java/com/codingblocks/cbonlineapp/adapters/ChoicesAdapter.kt (hunk 1 above)
Comment (raystatic) [reply app/src/main/java/com/codingblocks/cbonlineapp/adapters/ChoicesAdapter.kt]: this context will be required in bindViewHolder

File: app/src/main/java/com/codingblocks/cbonlineapp/fragments/QuizFragment.kt (hunk 6 above)
Comment (raystatic) [reply app/src/main/java/com/codingblocks/cbonlineapp/fragments/QuizFragment.kt]: this list is being used further

**Pull request closed without merging**